### PR TITLE
Fix tsh windows builds

### PR DESCRIPTION
**Pull request by rosstimothy**
https://github.com/gravitational/teleport/pull/24864 added a dependency of `lib/web` into tsh which broke windows tsh builds because `lib/web` transiently depends on `lib/srv` which has linux specific code. This shuffles around a few things so that `lib/web` is no longer importing `lib/srv` at all by:

1) Indirectly using the `srv.SessionController` to apply session control for web ssh sessions
2) Moving the common reversetunnel interfaces into `reversetunnelclient` since `lib/reversetunnel` imports `lib/srv/forward` which imports `lib/srv`.
3) Directly converting mysql client errors in the connection tester instead of calling a common function.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: lib/reversetunnel/api.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2016 Gravitational, Inc.
+Copyright 2020 Gravitational, Inc.
```
Comment (jakule) [inline lib/reversetunnel/api.go:2]: nit: `2023` or `2016-2023`

File: lib/reversetunnelclient/api.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2016 Gravitational, Inc.
```
Comment (jakule) [inline lib/reversetunnelclient/api.go:2]: nit
```suggestion
Copyright 2013 Gravitational, Inc.
```

File: lib/web/apiserver.go [hunk 3]
```diff
@@ -876,7 +875,7 @@ func (h *Handler) handleGetUserOrResetToken(w http.ResponseWriter, r *http.Reque
 // getUserContext returns user context
 //
 // GET /webapi/sites/:site/context
-func (h *Handler) getUserContext(w http.ResponseWriter, r *http.Request, p httprouter.Params, c *SessionContext, site reversetunnel.RemoteSite) (interface{}, error) {
+func (h *Handler) getUserContext(w http.ResponseWriter, r *http.Request, p httprouter.Params, c *SessionContext, site reversetunnelclient.RemoteSite) (interface{}, error) {
```
Comment (jakule) [inline lib/web/apiserver.go:878]: nit:
```suggestion
func (h *Handler) getUserContext(w http.ResponseWriter, r *http.Request, p httprouter.Params, c *SessionContext, site reversetunnelclient.RemoteSite) (any, error) {
```

File: lib/web/apiserver.go [hunk 4]
```diff
@@ -2624,6 +2589,24 @@ func (h *Handler) deleteClusterLock(
 	return OK(), nil
 }
 
+// SessionController restricts creation of sessions based on
+// cluster session control configuration(locks, connection limits, etc).
+type SessionController interface {
+	AcquireSessionContext(ctx context.Context, sctx *SessionContext, login, localAddr, remoteAddr string) (context.Context, error)
```
Comment (jakule) [inline lib/web/apiserver.go:2595]: IMO getting and returning `context.Context` is confusing. Can you at least mention in the comment what is being returned? Child context created from the passed parent? What will happen when I cancel it? Will the whole thing cancel?

**Review by jakule — APPROVED** [review]
Overall looks good

**Review by zmb3 — APPROVED** [review]

**Pull request merged**